### ISSUE TITLE
specification for the atomic swap communications side-channel?

### ISSUE THREAD

**Issue by markblundeberg** [decred/atomicswap#86]
I am thinking for the non-lightning crypto swaps, it would be great if the wider community could settle on some kind of standard for the side channel protocol. I'm not sure if there is any movement in this direction so I'm just leaving an issue post here to inquire.

One reason I ask is that I hope at some point there is a plugin for Electrum / Electron Cash that lets users make atomic swaps, and ideally it would interoperate with decred, and with swap tools that may appear for other wallets. I may even write such a plugin myself.

There is perhaps not any need to freeze a protocol yet but I hope that with your practical experience, something elegant could be settled on.

At the risk of self-promoting too much, I would hope the protocol also supported [other variants](https://gist.github.com/markblundeberg/ee027cb12b1e882145bebcd658a1630f) of atomic swap contract, too. And maybe even to permit swaps carrying OP_RETURN data like the [colored coins token system](https://simpleledger.cash/) we just implemented. :-D

Cheers!

**Comment by xorzero777** [comment]
Yep, sounds good as a first approximation . . . but I would like to do a plugin later. I may have less of a problem getting trust from the specific small set of users I have than is the general case.
